gptcache/similarity_evaluation/sequence_match.py

```diff
@@ -0,0 +1,99 @@
+from typing import Tuple, Dict, Any, List
+import numpy as np
+
+from gptcache.similarity_evaluation import SimilarityEvaluation
+from gptcache.embedding.base import BaseEmbedding
+
+def euclidean_distance_calculate(vec_l: np.array, vec_r: np.array):
+    return np.sum((vec_l - vec_r)**2)
+
+class SequenceMatchEvaluation(SimilarityEvaluation):
+    """
+    Evaluate sentence pair similarity using SequenceMatchEvaluation.
+
+    :param embedding_extractor: The embedding extractor used to obtain embeddings from the text content.
+    :type embedding_extractor: gptcache.embedding.base.BaseEmbedding
+    :param weights: List of weights corresponding to each sequence element for calculating the weighted distance.
+    :type weights: List[float]
+
+    Example:
+        .. code-block:: python
+
+            from gptcache.similarity_evaluation import SequenceMatchEvaluation
+            from gptcache.embedding import Onnx
+
+            embedding_extractor = Onnx()
+            weights = [0.5, 0.3, 0.2]
+            evaluation = SequenceMatchEvaluation(embedding_extractor, weights)
+
+            query = {
+                'question': 'USER: "foo2" USER: "foo4"',
+            }
+
+            cache = {
+                'question': 'USER: "foo6" USER: "foo8"',
+            }
+
+            score = evaluation.evaluation(query, cache)
+    """
+
+
+    def __init__(self, embedding_extractor: BaseEmbedding , weights: List[float]):
```

Comment:
`embedding_extractor` param should be str, and need a extra param `kwarg` to be use to init the embedding